### ISSUE TITLE
command with multiline string parses in R, but not in Rmd

### ISSUE THREAD

**Issue by ras44** [rstudio/rstudio#6495]
If I run the below command in an R-script, it parses it correctly and executes it.  If I run it in an R-Markdown code chunk, it does not parse it correctly, and hangs waiting for further input:

```
dbGetQuery(athena,"
SELECT
  some_field
FROM  (
  SELECT
    some_field
  FROM my_table a
)
GROUP BY 1
ORDER BY 1
")
```

In R-Markdown, I get:
```
> dbGetQuery(athena,
+ "
+ SELECT
+   some_field
+ FROM  (
+   SELECT
+     some_field
+   FROM my_table a
+ )
+ # athena
+ dbGetQuery(athena,"
+ SELECT
Error: unexpected symbol in:
"dbGetQuery(athena,"
SELECT"
```

Session info:
```
> xfun::session_info('rmarkdown')
R version 3.6.2 (2019-12-12)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core), RStudio 1.2.5033

Locale:
  LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
  LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
  LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

Package version:
  base64enc_0.1.3 digest_0.6.20   evaluate_0.14   glue_1.3.1      graphics_3.6.2  grDevices_3.6.2 highr_0.8      
  htmltools_0.3.6 jsonlite_1.6    knitr_1.25      magrittr_1.5    markdown_1.1    methods_3.6.2   mime_0.7       
  Rcpp_1.0.2      rmarkdown_1.16  stats_3.6.2     stringi_1.4.3   stringr_1.4.0   tinytex_0.16    tools_3.6.2    
  utils_3.6.2     xfun_0.10       yaml_2.2.0     

Pandoc version: 2.3.1

```

---
By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('rmarkdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/rmarkdown')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by ras44** [comment]
@kevinushey not sure if this issue has been addressed, or if it ought to be closed by the bot as noted above - thanks!

**Comment by ras44** [comment]
@ronblum @kevinushey Sorry, wasn't able to keep this active before it was closed.  I believe this should still be considered an open bug.

**Comment by Serenthia** [comment]
@ras44 Actually, the above holds out in just a normal .R file, so it's perhaps a more global issue?

**Comment by ras44** [comment]
oh, interesting.  I'm now using R Studio Server 1.4.1103 on Ubuntu and with this version I'm unable to reproduce.  I can put multilineString2 into an RMarkdown code chunk and ctrl-Enter through the four lines (only requires pushing ctrl-Enter three times as two lines are parsed together on one occasion).  In an R script, I only have to press ctrl-Enter once and it will parse the entire string correctly.

The original example I described appears to have been resolved:

In an R-Markdown document I can enter:
```
print("
SELECT
  some_field
FROM  (
  SELECT
    some_field
  FROM my_table a
)
GROUP BY 1
ORDER BY 1
")
```
And it will require me to ctrl-Enter four times, as it stops parsing after the first ')'.  However it does not produce the "Error: unexpected symbol in:" error as it did when this issue was created.  In an R script, it parses the entire string with just one ctrl-Enter.

I would consider this specific issue resolved as of R 1.4.1103, however you may want to open a new issue if the multi-line string with an empty line still produces an error.

If it's ok with you @Serenthia I'll resolve this one.
